src/test/integration/storage/test_db_interface_frontend.py

```diff
@@ -46,28 +47,39 @@ def test_get_hid(db):
 
 
 def test_get_hid_fo(db):
-    test_fo = create_test_file_object(bin_path='get_files_test/testfile2')
-    test_fo.virtual_file_path = {'a': ['|a|/test_file'], 'b': ['|b|/get_files_test/testfile2']}
-    db.backend.insert_object(test_fo)
-    result = db.frontend.get_hid(test_fo.uid, root_uid='b')
+    fo, parent_1, fw_1, fw_2 = get_fo_with_2_root_fw()
+    fo.virtual_file_path = {parent_1.uid: ['/test_file'], fw_2.uid: ['/get_files_test/testfile2']}
+    db.backend.insert_multiple_objects(fw_2, fw_1, parent_1, fo)
+    result = db.frontend.get_hid(fo.uid, root_uid=fw_2.uid)
     assert result == '/get_files_test/testfile2', 'fo hid not correct'
-    result = db.frontend.get_hid(test_fo.uid)
+    result = db.frontend.get_hid(fo.uid)
     assert isinstance(result, str), 'result is not a string'
-    assert result[0] == '/', 'first character not correct if no root_uid set'
-    result = db.frontend.get_hid(test_fo.uid, root_uid='c')
-    assert result[0] == '/', 'first character not correct if invalid root_uid set'
+    assert result.startswith('/'), 'first character not correct if no root_uid set'
+    result = db.frontend.get_hid(fo.uid, root_uid='invalid')
+    assert result == fo.file_name, 'file name should be fallback'
 
 
 def test_get_hid_invalid_uid(db):
     result = db.frontend.get_hid('foo')
     assert result == '', 'invalid uid should result in empty string'
 
 
+def test_get_hid_dict(db):
+    fo, fw = create_fw_with_child_fo()
+    fo.virtual_file_path[fw.uid] = ['/foo']
+    db.backend.insert_multiple_objects(fw, fo)
+    uid_set = {fo.uid, fw.uid}
+    hid_dict = db.frontend.get_hid_dict(uid_set, root_uid=fw.uid)
+    assert all(uid in hid_dict for uid in uid_set)
+    assert hid_dict[fo.uid] == '/foo'
+    assert hid_dict[fw.uid] == f'{fw.vendor} {fw.device_name} - {fw.version} ({fw.device_class})'
```

Comment:
This is the "firmware HID" (without "part")

> Does a helper function doing this string formatting exist?

Yes, `_get_hid_firmware()`, but since it is a "private" helper function, it is not tested individually (but `get_hid()` is tested).
=> replaced it with something less explicit